packages/mobile/src/account/ConnectPhoneNumberScreen.tsx

```diff
@@ -17,6 +21,14 @@ function onPressConnectButton() {
 export default function ConnectPhoneNumberScreen() {
   const { t } = useTranslation()
 
+  // Log a cancel event on a "back" action (hardware back button, swipe, or normal navigate back)
+  const navigation = useNavigation()
+  React.useEffect(() => {
+    navigation.addListener('beforeRemove', (e) => {
+      ValoraAnalytics.track(CICOEvents.link_bank_account_cancel)
+    })
+  })
```

Comment:
We should always pass the dependencies array (here empty is fine as `navigation` doesn't change), otherwise it will apply the effect on each re-render.
Also it's necessary to remove the listener when unmounting. This is done by returning the result from `addListener`. See https://reactnavigation.org/docs/navigation-events/#navigationaddlistener
```suggestion
  React.useEffect(
    () =>
      navigation.addListener('beforeRemove', (e) => {
        ValoraAnalytics.track(CICOEvents.link_bank_account_cancel)
      }), 
    []
  )
```